app/(auth)/index.tsx

```diff
@@ -0,0 +1,166 @@
+import { Link, router, useFocusEffect } from 'expo-router';
+import { useCallback, useState } from 'react';
+import {
+  Alert,
+  Pressable,
+  SafeAreaView,
+  StyleSheet,
+  Text,
+  TextInput,
+  View,
+} from 'react-native';
+import { colors } from '../../constants/colors';
+import type { LoginResponseBodyPost } from '../api/login+api';
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    backgroundColor: colors.background,
+    alignItems: 'center',
+    width: '100%',
+  },
+  loginInputContainer: {
+    backgroundColor: colors.cardBackground,
+    borderRadius: 12,
+    padding: 12,
+    marginBottom: 16,
+    width: '100%',
+  },
+  label: {
+    fontSize: 18,
+    fontFamily: 'Poppins_400Regular',
+    color: colors.text,
+    marginBottom: 8,
+  },
+  input: {
+    color: colors.text,
+    backgroundColor: colors.background,
+    borderColor: colors.textSecondary,
+    borderWidth: 1,
+    borderRadius: 8,
+    padding: 8,
+    marginBottom: 16,
+    fontSize: 16,
+    fontFamily: 'Poppins_400Regular',
+  },
+  inputFocused: {
+    borderColor: colors.white,
+  },
+  promptTextContainer: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
+    gap: 4,
+  },
+  button: {
+    marginTop: 30,
+    backgroundColor: colors.text,
+    padding: 12,
+    borderRadius: 12,
+    shadowColor: colors.white,
+    shadowOffset: { width: 0, height: 2 },
+    shadowOpacity: 0.2,
+    shadowRadius: 4,
+    elevation: 4,
+    width: '100%',
+  },
+  text: {
+    fontFamily: 'Poppins_400Regular',
+    color: colors.cardBackground,
+    textAlign: 'center',
+    fontSize: 18,
+  },
+});
+
+export default function Login() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [focusedInput, setFocusedInput] = useState<string | undefined>();
+
+  useFocusEffect(
+    useCallback(() => {
+      async function getUser() {
+        const response = await fetch('/api/user');
+
+        const data = await response.json();
+
+        if ('username' in data) {
+          router.push('/(tabs)/guests');
+        }
```

Comment:
It appears that the `?returnTo` functionality is missing here

Reply:
495808f5ef6f50803cf2efbf79ba88cd5209ae05
672bdbb8da45e70f7e832520da35680df82f01e9

Reply:
Can we use `?returnTo=/(tabs)/guests` instead of `?returnTo=/guests` ?

Then we can do this in the login:

```ts
if (returnTo && typeof returnTo === 'string') {
  router.push(returnTo as Href);
}
```

Reply:
Done in [`9a6951e` (#1)](https://github.com/upleveled/expo-example-fall-2024-atvie/pull/1/commits/9a6951e3f17b69dcd5cb128ab18c583ffec492f4)